source/common/common/mutex_contention.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <atomic>
+
+namespace Envoy {
+namespace Thread {
+
+namespace {
+std::atomic<int64_t> num_contentions(0);
+std::atomic<int64_t> current_wait_cycles(0);
+std::atomic<int64_t> lifetime_wait_cycles(0);
```

Comment:
can we put the atomics in a struct?  that would avoid static initialized atomics which arguably violate style guide. A single static struct pointer would hold the allocation which would occur on registration, and the nullness of the pointer would help us avoid extra calls to registration.

Reply:
Flushing comments as out of date.